wpilibNewCommands/src/main/native/cpp/frc2/command/CommandPtr.cpp

```diff
@@ -167,6 +167,34 @@ CommandPtr CommandPtr::RaceWith(CommandPtr&& parallel) && {
   return std::move(*this);
 }
 
+class ThenCommand : public WrapperCommand {
```

Comment:
Rename to FinallyDoCommand? Also, since this is an implementation class, it should be wrapped in `namespace { ... }` (anonymous namespace).